edb/testbase/server.py

```diff
@@ -114,9 +114,11 @@ def wrapper(self, *args, __meth__=meth, **kwargs):
                     if try_no == 3:
                         raise
                     else:
-                        self.loop.run_until_complete(self.con.execute(
-                            'ROLLBACK;'
-                        ))
+                        if getattr(self, 'ISOLATED_METHODS'):
```

Comment:
Perhaps condition the entire loop in this `getattr`?  It probably makes no sense to retry tests that we haven't implicitly established a transaction for.